null_bool.go

```diff
@@ -0,0 +1,103 @@
+package dosa
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strconv"
+)
+
+// NullBool is a type that holds an optional boolean value.
+type NullBool struct {
+	sql.NullBool
+}
+
+// NewNullBool returns a new NullBool object that is initialized to the
+// specified value.
+func NewNullBool(value bool) NullBool {
+	return NullBool{
+		NullBool: sql.NullBool{
```

Comment:
nit: `NullBool:` is optional. While it is the name, this is an anonymous member.